Quickstart updates (#74)
auth0/docs-v2: auth4genai/snippets/get-started/langchain-fastapi-py/call-others-api.mdx

  }}
/>

<Tabs>
  <Tab
    title="Use sample app (recommended)"
  >
  
### Clone sample app
Clone this sample app from the [Auth0 AI samples](https://github.com/auth0-samples/auth0-ai-samples) repository:

```bash wrap lines
git clone https://github.com/auth0-samples/auth0-ai-samples.git
cd auth0-ai-samples/call-apis-on-users-behalf/others-api/langchain-fastapi-py
```

The project is divided into two parts:

- `backend/`: contains the backend code for the web app, an API written in Python using FastAPI and the LangGraph agent.
- `frontend/`: contains the frontend code for the web app, written in React as a Vite SPA.

### Install backend packages

> In the `backend` directory of your project, install the required packages using your preferred package manager, such as `uv`. You can follow the instructions here to [install uv](https://docs.astral.sh/uv/getting-started/installation/).

Reviewer: ```suggestion
In the `backend` directory of your project, install the required packages using your preferred package manager, such as `uv`. You can find the instructions on how to [install uv](https://docs.astral.sh/uv/getting-started/installation/) in its documentation.
```

uv sync
```

### Create your environment file

In the `backend` directory of your project, create a new file and name it `.env` and add the following content:

```bash .env wrap lines
# Auth0 configuration
APP_BASE_URL='http://localhost:8000'
AUTH0_SECRET='random 32 byte value'
AUTH0_DOMAIN='<your-auth0-domain>'
AUTH0_CLIENT_ID='<your-auth0-application-client-id>'
AUTH0_CLIENT_SECRET='<your-auth0-application-client-secret>'

# OpenAI API Key or any provider supported by the Vercel AI SDK
OPENAI_API_KEY="YOUR_API_KEY"

# LANGGRAPH
LANGGRAPH_API_URL=http://localhost:54367
```

To get your Auth0 application's `AUTH0_DOMAIN`, `AUTH0_CLIENT_ID`, and `AUTH0_CLIENT_SECRET`, navigate to the **Settings** tab in the Auth0 Dashboard. You'll find these values in the **Basic Information** section at the top.
Copy each value to the matching setting.

Next, run this command to generate a random 32 byte value and copy it to the `AUTH0_SECRET` field:
```bash generate random 32 byte value
openssl rand -hex 32
```

Lastly, set your OpenAI API key or use any provider supported by the Vercel AI SDK:
```bash .env.local wrap lines
OPENAI_API_KEY="YOUR_API_KEY"
```

### Install frontend packages

> Ensure you have `npm` installed or follow the instructions here to [install npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm).

Reviewer: ```suggestion
Ensure you have `npm` installed or follow the instructions to [install npm](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm) in its documentation.
```

Navigate to the `frontend` directory of your project and install the required packages:

```bash wrap lines
cd frontend
npm install
```

### Run your application

To run your application, start the FastAPI backend, LangGraph server, and the frontend in three terminals:

1. In one terminal, start the FastAPI backend:

```bash wrap lines
cd backend
source .venv/bin/activate
uv pip install auth0_fastapi
fastapi dev app/main.py
```

2. In a second new terminal, start the LangGraph server:

```bash wrap lines
cd backend
source .venv/bin/activate
uv pip install -U langgraph-api
langgraph dev --port 54367 --allow-blocking
```
<Note>
  This opens LangGraph Studio in a new tab. You can safely close it.
</Note>

3. In a third new terminal, start the frontend:

```bash wrap lines
cd frontend
cp .env.example .env # Copy the `.env.example` file to `.env`.
npm install
npm run dev
```

> Visit the URL `http://localhost:5173` in your browser. If you are already logged in, make sure to log out and log back in using Google. Now ask your AI Agent to list the upcoming events in your Google Calendar!

Reviewer: ```suggestion
Visit the URL `http://localhost:5173` in your browser. If you are already logged in, make sure to log out and log back in using Google. Now ask your AI agent to list the upcoming events in your Google Calendar!
```

> That's it! You've successfully integrated third-party API access using Token Vault into your LangGraph FastAPI project.

Reviewer 2: Maybe slightly different verbage than "integrated" at the end here (since we are in the"Use sample app" tab).

### Test your application

Now when step-up authorization is required, the user will see a prompt in the chat window asking them to authorize.

> ### Test your application

Reviewer 2: Currently attempting to reflect this style w/ a new quickstart here 😸 . Looking good!

Just noting "Test your application" used twice here ^^
